app/controllers/api/base_controller/authentication.rb

```diff
@@ -7,11 +7,13 @@ module Authentication
       # REST APIs Authenticator and Redirector
       #
       def require_api_user_or_token
+        using_http_basic = false
```

Comment:
Shouldn't this be a method?

```ruby
def using_http_basic_auth?
  !(request.headers[HttpHeaders::MIQ_TOKEN] || request.headers[HttpHeaders::AUTH_TOKEN])
end
```

Reply:
or `def auth_mechanism` or something

Reply:
OK, makes sense :)

Will do, looks like there is a similar logic in container deployments, I'll try to merge that into a helper.

```
app/controllers/api/container_deployments_controller.rb
20:      [HttpHeaders::MIQ_TOKEN, HttpHeaders::AUTH_TOKEN, "HTTP_AUTHORIZATION"].any? do |header|
```